Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -1716,63 +1673,33 @@ void IMainLoopDriver.Setup (MainLoop mainLoop)
         Task.Run (NetInputHandler, _inputHandlerTokenSource.Token);
     }
 
-    void IMainLoopDriver.Wakeup () { _eventReady.Set (); }
+    void IMainLoopDriver.Wakeup () {  }
 
     bool IMainLoopDriver.EventsPending ()
     {
         _waitForProbe.Set ();
 
-        if (_mainLoop.CheckTimersAndIdleHandlers (out int waitTimeout))
-        {
-            return true;
-        }
-
-        try
-        {
-            if (!_eventReadyTokenSource.IsCancellationRequested)
-            {
-                // Note: ManualResetEventSlim.Wait will wait indefinitely if the timeout is -1. The timeout is -1 when there
-                // are no timers, but there IS an idle handler waiting.
-                _eventReady.Wait (waitTimeout, _eventReadyTokenSource.Token);
```

Comment:
Did you realized that the `EventsPending` method is always running overkilling the CPU usage with 13%, approximately? That why the `Threading` scenario it's working on `Linux` and `Windows`. It's needed to wait until it's necessary. So, the `BlockingCollection` is only blocking in the `NetInputHandler` method but the `EventsPending` method is always running. I think the `BlockingCollection` blocks in it own thread and cause the driver not responding at all hanging in `Linux`. In `Windows` may have some managed code that can surpass this issue, maybe because `NetDriver` is using Win32 API which respond to a focus event, I don't know.

Reply:
Thanks for pointing this out, I have fixed in [49475b6](https://github.com/gui-cs/Terminal.Gui/pull/3791/commits/49475b6f9481e23ad0e1375edc4f9b4a80fe8217)